sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -245,6 +245,35 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
       }.toArray
       DeleteFromTableExec(r.table.asDeletable, filters) :: Nil
 
+    case UpdateTable(r: DataSourceV2Relation, attrs, values, condition) =>
+      if (condition.exists(SubqueryExpression.hasSubquery)) {
+        throw new AnalysisException(
+            s"Update by condition with subquery is not supported: $condition")
+      }
+      val nested = attrs.filterNot(_.isInstanceOf[AttributeReference])
+      if (nested.nonEmpty) {
```

Comment:
We don't need this now. The check is already done below.

Reply:
Thank you @cloud-fan , keen sight!